plugin/storage/es/mappings/mapping.go

```diff
@@ -9,36 +9,70 @@ import (
 	"strings"
 
 	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/pkg/es/config"
 )
 
 // MAPPINGS contains embedded index templates.
 //
 //go:embed *.json
 var MAPPINGS embed.FS
 
-// MappingBuilder holds parameters required to render an elasticsearch index template
+// MappingBuilder holds common parameters required to render an elasticsearch index template
 type MappingBuilder struct {
-	TemplateBuilder              es.TemplateBuilder
-	Shards                       int64
-	Replicas                     int64
-	PrioritySpanTemplate         int64
-	PriorityServiceTemplate      int64
-	PriorityDependenciesTemplate int64
-	PrioritySamplingTemplate     int64
-	EsVersion                    uint
-	IndexPrefix                  string
-	UseILM                       bool
-	ILMPolicyName                string
+	TemplateBuilder es.TemplateBuilder
+	Indices         config.Indices
+	EsVersion       uint
+	UseILM          bool
+	ILMPolicyName   string
+}
+
+// IndexTemplateOptions holds parameters required to render an elasticsearch index template
+type IndexTemplateOptions struct {
+	UseILM        bool
+	ILMPolicyName string
+	IndexPrefix   config.IndexPrefix
```

Comment:
I think this should be a string. Semantically it's not the same as IndexPrefix, it's the full prefix+indexname, is it not?

Reply:
@yurishkuro The reasoning for this was that we were doing this in fixMapping:
```go
	if options.IndexPrefix != "" && !strings.HasSuffix(options.IndexPrefix, config.IndexPrefixSeparator) {
		options.IndexPrefix += config.IndexPrefixSeparator
	}
```
which could be replaced with 
```go
	options.IndexPrefix = config.IndexPrefix(options.IndexPrefix.Apply(""))
```
so it seems to have similar semantics to the other usages. What do you think?

Reply:
> `options.IndexPrefix = config.IndexPrefix(options.IndexPrefix.Apply(""))`

+1, but do you later do `options.IndexPrefix.Apply()` again? If yes then ok, it can be typed string.